### PR TITLE
Ce/enterprise mode

### PR DESCRIPTION
**Pull request by calellowitz**
@czue @sravfeyn 
First pass at a custom landing page for icds, and the infrastructure to do this more. Pending approval from NIC on the template design. It will also require some localsettings changes in ansible which I can make after approval.

### CODE REVIEW

File: corehq/apps/hqwebapp/views.py [hunk 1]
```diff
@@ -292,7 +292,17 @@ def _login(req, domain_name, template_name):
     req.base_template = settings.BASE_TEMPLATE
 
     context = {}
-    if domain_name:
+    default_page_name = _('Welcome back to CommCare HQ!')
+    if settings.CUSTOM_LANDING_PAGE:
+        page_name = getattr(settings, 'CUSTOM_WELCOME_MESSAGE', default_page_name)
+        custom_template =  getattr(settings, 'CUSTOM_LANDING_TEMPLATE', False)
```
Comment (dimagimon) [inline corehq/apps/hqwebapp/views.py]: E222 multiple spaces after operator

**Review by czue — APPROVED** [review]
Looks good. Is the logo already in the repo or how is that going to work?

**Comment by calellowitz** [comment]
I got it to work locally by putting in the address for the icds-cas logo on the india server (apparently static assets dont require auth...) but once the domain is migrated and the custom logo is set on this server we can just pull it from the domain in the icds cluster.

**Comment by czue** [comment]
real build fails
```
======================================================================
ERROR: test_csrf_ON (corehq.apps.hqwebapp.tests.test_csrf_middleware.TestCSRF)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/mnt/commcare-hq-ro/corehq/apps/hqwebapp/tests/test_csrf_middleware.py", line 26, in test_csrf_ON
    csrf_sent, csrf_missing = self._form_post_with_and_without_csrf()
  File "/mnt/commcare-hq-ro/corehq/apps/hqwebapp/tests/test_csrf_middleware.py", line 32, in _form_post_with_and_without_csrf
    login_page = client.get(reverse('login'))
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 500, in get
    **extra)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 303, in get
    return self.generic('GET', path, secure=secure, **r)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 379, in generic
    return self.request(**r)
  File "/vendor/lib/python2.7/site-packages/django/test/client.py", line 466, in request
    six.reraise(*exc_info)
  File "/vendor/lib/python2.7/site-packages/django/core/handlers/base.py", line 132, in get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/vendor/lib/python2.7/site-packages/django/views/decorators/debug.py", line 76, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/hqwebapp/views.py", line 331, in login
    return _login(req, domain, "login_and_password/login.html")
  File "/vendor/lib/python2.7/site-packages/django/views/decorators/debug.py", line 76, in sensitive_post_parameters_wrapper
    return view(request, *args, **kwargs)
  File "/mnt/commcare-hq-ro/corehq/apps/hqwebapp/views.py", line 296, in _login
    if settings.CUSTOM_LANDING_PAGE:
  File "/vendor/lib/python2.7/site-packages/django/conf/__init__.py", line 49, in __getattr__
    return getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'CUSTOM_LANDING_PAGE'
```

File: corehq/util/context_processors.py [hunk 2]
```diff
@@ -85,3 +85,8 @@ def websockets_override(request):
         # it's very unlikely this was needed, and some workflows (like scheduled reports) aren't
         # able to generate this, so don't worry about it.
         return {}
+
+def enterprise_mode(request):
```
Comment (dimagimon) [inline corehq/util/context_processors.py]: E302 expected 2 blank lines, found 1

**Comment by calellowitz** [comment]
@czue @snopoke @sravfeyn 
I added the rest of enterprise mode to this PR. Its currently just hiding options from menus as per https://docs.google.com/document/d/1uw2Sv7WIEC0xiHzjM63IsJ7HIIoVZh2sIPDKiIBjhlk/edit

**Comment by snopoke** [comment]
looks good. Do we need to add loop holes in any of these places for super users? I don't think so but just checking.

**Comment by calellowitz** [comment]
Im not sure we need the exchange at all. The admin accounting interface can be used to handle subscriptions if we need that. I also dont think we will need to create new domains but if so the direct url to get to the domain creation form should work.

**Comment by snopoke** [comment]
👍

**Pull request merged**